### PR TITLE
Admin UI : Hint dashes (—)

### PR DESCRIPTION
**Pull request by scleaver**
Removing the hard coded dashes from admin UI and replacing them with CSS.

Don't merge yet, this is not completed as I am seeking comment on this PR.

One thing that is quite annoying when you want to customize the admin UI is that dashes - and long dash — is used (interchangeably actually) with the hint and text muted making it almost impossible to remove them if you don't want that in your UI or your admin layout positions the hints differently.

Also, I don't know if that character is universal when it comes to multilingual use.

The approach I am suggesting with this PR is to remove all relevant use of dash in code and instead use CSS to insert the dash in TheAdmin template in the appropriate places using appropriate selectors:

`.switch ~ .hint::before  { content: "-" }`

So far in this PR I have removed all such occurrences and now I am beginning to make sure every use case has a CSS rule that re-inserts the dash back into the UI in the appropriate place.

I will try to write as few style rules as possible to cover the requirements.

What are the thoughts from other contributors? I think this should be done before official release.

### CODE REVIEW

**Comment by Skrypt** [comment]
I totally agree on the PR. Though I think the appropriate char is "—".

File: src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss
Comment (Skrypt) [inline src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: I don't think we need this selector. It can be applied everywhere we have a hint.

Comment (scleaver) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: @Skrypt - hints on TextField and other such fields don't have – at all which is why I was going through being more specific with the rule. Unless you just want me to add a modifier like .hint--dash and just add the modifier to the relevant templates and then do:

`.hint--dash::before { content: "–" }`

Comment (Skrypt) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: At some point we included the "double dash" in the translation files and we started removing them because it was not necessary to be translated. I think we should have a dash everywhere we use a .hint unless you say that it would look awkward in some place.

Comment (Skrypt) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: No one has taken this task so far because you will also need to remove these in the localization.po files.

Comment (scleaver) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: How about I try it just on .hint and see what it looks like. By the way, I can't see a gulp, grunt or npm command in TheAdmin theme...  how is the scss compiled?

Comment (Skrypt) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: It's built from the main `gulp build` root command. If it doesn't compile, you need to make a change in the TheAdmin.scss file for the gulp task to pick it up "small issue I know".

Comment (scleaver) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: The hint is used in a lot of places. Adding it simply to .hint adds it into the media picker and everything:

screencapture-localhost-5001-Admin-Contents-ContentTypes-Article-Create-2020-10-21-17_12_50

Comment (scleaver) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: And even here... personally I think "Last modified 2 minutes ago by admin" should actually have it's own class rather than re-using hint even if it a hint reused the same text styles.

Screen Shot 2020-10-21 at 5 24 59 pm

Comment (Skrypt) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: Ok then we need a class to specify which one should be dashed and we need to redo them one by one.
`<span class="hint dashed">hello</span>`
`.hint.dash::before { content: "—" }`

Comment (scleaver) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: Oh boy.

Comment (Skrypt) [reply src/OrchardCore.Themes/TheAdmin/Assets/scss/bootstrap-switch.scss:177]: 🙈 🙉 🙊 Time to go sleep for me 😉

**Comment by netwavebe** [comment]
I removed one a few days ago to make it more consistent with other fields: https://github.com/OrchardCMS/OrchardCore/pull/7382

The PredefinedList editor was the only one with a dash in front of the hint.

In my opinion, hints that are on it's own line, like below an editor, don't need a dash at all.
Hints next to a header or text DO need a dash.

**Comment by Skrypt** [comment]
There is a bunch of them everywhere used with and without a double dash. No consistency at all. I wish I had time to work on this but ... it's never a priority.

**Comment by scleaver** [comment]
@Skrypt - how do the po files work exactly as it is a little confusing? You have an example like this below in OrchardCore.Taxonomies.po - the localised text is only "Whether leaves are the only items that can be selected." and the — is actually outside the translated text so I am confused why the Arabic is including the - at the start.

`#. <span class="hint">— @T["Whether leaves are the only items that can be selected."]</span>
#: OrchardCore.Taxonomies\Views\TaxonomyFieldSettings.Edit.cshtml:53
msgctxt "OrchardCore.Taxonomies.Views.TaxonomyFieldSettings.Edit"
msgid "Whether leaves are the only items that can be selected."
msgstr "- ما إذا كانت الأطراف هي العناصر التي يمكن اختيارها."
`
Any light you can shed would be tops.

**Comment by Skrypt** [comment]
Yeah on top of that some .po files are not up to date with latest code. Like I said, at some point in time the double dashes were included in the localizable string.

**Comment by Skrypt** [comment]
I would make a search of `class="hint"` and then replace it with `class="hint dashed"` everywhere it should be by doing them one by one and analyzing the localizable string next to it.

Afterward I would find and replace globally on every `—` that I could find in the view templates.
Then finally, replace them in all the .po files downloaded in the OrchardCore.Cms.Web/Localization folder. Then submit them on Crowdin.

Everything that could be usefull to clean some of them. Then finally

**Comment by scleaver** [comment]
There are some like this `class="text-muted">- ContentLink</span>` with a single dash. Should I replace these with a dash or a long dash also (using CSS)?

**Comment by scleaver** [comment]
Done.. accept I accidentally included some commits for a yaml for Azure DevOps when I was mucking around with builds on my fork :( - Looks like there is major changes to the admin theme so I will need to sort that too.

**Comment by scleaver** [comment]
Ok the UI changes have been done and the CSS has been added and this should be good to go. @Skrypt - with the .po files I am not sure how I upload these into Crowdin but I am assuming I don't commit the changes here.

**Comment by scleaver** [comment]
@Skrypt - will have to try to get this merged soon because it is hard to keep it up with all the changes that are happening in AdminTheme and other modules.

Also need to find out about how I upload the .po files into Crowdin

**Comment by agriffard** [comment]
> Also need to find out about how I upload the .po files into Crowdin

I will extract and update the .po files once it is merged.

**Review by deanmarcussen — COMMENTED** [review]
@Skrypt can you review this?

Not sure why it has the do not merge tag.

Would like to get the merge conflicts (css rebuilt) resolved, and this pr in before it goes stale.

Also there are some odd looking changes to the blog theme js.

(concept good though)

**Comment by scleaver** [comment]
@deanmarcussen and @Skrypt - not sure why that happened with the blog theme. After I did a gulp build on the latest that was the result. looks like there are no actual changes though.

I have fixed the conflicts with this again this morning.

**Comment by Skrypt** [comment]
I'm sorry guys I had a bad headache day. I needed some time off.

**Comment by Skrypt** [comment]
Ok guys try and merge latest dev branch and do a `gulp build`.

**Review by Skrypt — CHANGES_REQUESTED** [review]
Ok guys try and merge latest dev branch and do a gulp build to remove these rebuild of assets that are not necessary.

**Comment by Skrypt** [comment]
Probably my fault because of the gulpfile change I made. If he merge latest and rebuild that should fix it.

**Comment by deanmarcussen** [comment]
@scleaver can you find the commit where these blog theme files were changed and revert that part of the commit?

I'm not sure why they are showing empty (maybe it's just empty changes), but those files shouldn't be part of this pr whether they have changes or not

**Comment by scleaver** [comment]
@deanmarcussen and @Skrypt ok I have reverted those changes and it should be good now and squeaky clean like your were after dean. Also up to date with latest.

**Comment by deanmarcussen** [comment]
Ah so close. But another merge conflict.

This has my approval when the merge conflict is resolved.

Please merge it before https://github.com/OrchardCMS/OrchardCore/pull/7438

on the basis that older prs should win on the merge conflict battle...

**Comment by Skrypt** [comment]
Yeah we can't push commits on top of @scleaver private repo so as soon as it gets fixed we merge it.

**Comment by scleaver** [comment]
@Skrypt @deanmarcussen ok conflicts sorted. Let's get this closed.

**Pull request merged**
